projects/plugins/super-cache/tests/e2e/lib/plugin-settings.ts

```diff
@@ -50,7 +50,7 @@ const settingsHandlers = {
 };
 
 type SettingName = keyof typeof settingsHandlers;
-type SettingMethod< Name extends SettingName > = typeof settingsHandlers[ Name ];
+type SettingMethod< Name extends SettingName > = ( typeof settingsHandlers )[ Name ];
```

Comment:
Is this a correct change? I read the old version (perhaps naively) as checking the type of the value at the `Name` key.

Reply:
I don't really know TypeScript, so I couldn't really say. But it's possible that it's getting the type of the whole object, then extracting the type of the Name field?

Reply:
I'm not a TypeScript guru either, but if I'm right, it'll probably cause errors (albeit just in a test), since it'd be grabbing the Name property of the result of the `typeof` statement. Does this test currently run anywhere? I didn't see it in the E2E tests.

Reply:
No idea. OTOH, I see similar `( typeof something )[ key ]` in several other files, e.g.

https://github.com/Automattic/jetpack/blob/eafa13ac3d3021a9c561c9b11736b30c651d2d52/projects/js-packages/ai-client/src/types.ts#L89
https://github.com/Automattic/jetpack/blob/eafa13ac3d3021a9c561c9b11736b30c651d2d52/projects/js-packages/components/components/layout/use-breakpoint-match/index.ts#L6
https://github.com/Automattic/jetpack/blob/eafa13ac3d3021a9c561c9b11736b30c651d2d52/projects/js-packages/social-logos/src/react/social-logo.tsx#L6